### ISSUE TITLE
Two identical expressions are reported as not equal

### ISSUE THREAD

**Issue by mpickering** [nomeata/inspection-testing#8]
Running `inspection-testing` on a recent example in `generic-lens` reports a strange result.

```
    LHS:
        lvl_sjI5
        lvl_sjI5
          = \ s1 ->
              case s1 of wild_X2X {
                __DEFAULT -> Left wild_X2X;
                B i -> Right i
              }
        
        sum1PrismManual
        sum1PrismManual
          = \ @ p $dChoice_acxd eta ->
              dimap
                ($p1Choice $dChoice_acxd)
                lvl_sjI5
                lvl_sr6L
                (right' $dChoice_acxd eta)
        
    RHS:
        lvl_sr6K
        lvl_sr6K
          = \ x ->
              case x of wild_X1S {
                __DEFAULT -> Left wild_X1S;
                B g1 -> Right g1
              }
        
        sum1PrismB
        sum1PrismB
          = \ @ p $dChoice_acjG eta ->
              dimap
                ($p1Choice $dChoice_acjG)
                lvl_sr6K
                lvl_sr6L
                (right' $dChoice_acjG eta)
```

Could it be that `inspection-testing` is getting confused by the fact that `lvl_sr6K` and `lvl_sjI5` are not CSEd? It certainly looks like the two expressions optimise to the same thing.

**Comment by mpickering** [comment]
It works with 0.1.2 sorry for the noise.
